R/generate_dust.R

```diff
@@ -177,10 +176,10 @@ generate_dust_core_size <- function(dat, rewrite) {
     cpp_function("size_t", "size", NULL, body)
   } else {
     body <- sprintf("return %s;", rewrite(dat$data$variable$length))
-    n_var <- cpp_function("size_t", "n_variables", NULL, body)
+    n_var <- cpp_function("size_t", "n_variables", NULL, body, TRUE)
```

Comment:
Could you just explain what the introduction of this `const` argument is about?

Reply:
this changes the method

```c++
size_t n_variables() {
  return <some number>;
}
```

with

```c++
size_t n_variables() const {
  return <some number>;
}
```

that means that `n_variables` can be called from other const methods, which mode now does:

```c++
  size_t n_state_full() const {
    return m_.n_variables() + m_.n_output();
  }
```

Without the `const` added above we can't call the second function because we pass in an immutable pointer to the object and then try to call a method that is not explicitly immutable. The rust default is much better

Reply:
thanks